### PR TITLE
update permissions help

### PR DESCRIPTION
**Pull request by alxnddr**
Epic https://github.com/metabase/metabase/issues/29491

### Description

Update permissions help sidebar content.

### How to verify

1. As an admin go to http://localhost:3000/admin/permissions
2. Check data permissions help
3. Check collection permissions help
4. Check application permissions help (EE-only)

### Demo

Screen Shot 2023-06-07 at 9 56 41 PM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by alxnddr** [comment]
@vbenedetti @luizarakaki on the mockups we add `(Pro)` to Impersonated and Sandboxed permissions but not to the Block permission which is also only for EE users. Should we add it there as well?

Also the phrase "Only available in certain Metabase plans." that goes after the Block permission without the upgrade link and after Sandboxed with the upgrade link. The Impersonated permission description does not have this phrase — should it be there?

Screen Shot 2023-06-08 at 11 08 43 PM

File: frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+              icon="eye_crossed_out"
+              iconColor="accent5"
+              name={t`No self-service`}
+              description={t`The group can't use the query builder or drill through existing questions. They also can't see the data in the Browse data section. They can still view questions based on this data, if they have permissions to the relevant collection.`}
+            />
+
+            <PermissionHelpDescription
+              icon="close"
+              iconColor="danger"
+              name={t`Block (Pro)`}
```
Comment (JesseSDevaney) [inline frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx:57]: I agree that block should have a (Pro) descriptor, but it was not there in the design docs. I assume you confirmed this with product already?
```suggestion
              name={t`Block`}
```

File: frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            />
+
+            <PermissionHelpDescription
+              icon="permissions_limited"
+              iconColor="brand"
+              name={t`Sandboxed (Pro)`}
+              description={t`Let's you specify row and column-level permissions. Can be set up via user attributes and SSO.`}
+            />
+
+            <Text>{t`Only available in certain Metabase plans.`}</Text>
```
Comment (JesseSDevaney) [inline frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx:89]: I saw the design docs specified having an `Upgrade to Pro` link. Were you waiting to add this to see if product wanted it on both accordions under 'Only available in certain Metabase plans'?

**Review by JesseSDevaney — APPROVED** [review]
LGTM - I just have two questions about some differences between the design doc and implementation

File: e2e/test/scenarios/permissions/admin-permissions.cy.spec.js [hunk 3]
```diff
@@ -734,3 +734,31 @@ describeEE("scenarios > admin > permissions", () => {
...
+
+describe("scenarios > admin > permissions", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("shows permissions help", () => {
+    cy.visit("/admin/permissions");
+    cy.get("main").within(() => {
```
Comment (npfitz) [inline e2e/test/scenarios/permissions/admin-permissions.cy.spec.js:746]: Is there a more specific selector we can use here?

**Comment by luizarakaki** [comment]
@alxnddr Yes for both

- Please add `(Pro)` for block.
- Add the upgrade link after Block and Impersonated

File: e2e/test/scenarios/permissions/admin-permissions.cy.spec.js (hunk 3 above)
Comment (alxnddr) [reply e2e/test/scenarios/permissions/admin-permissions.cy.spec.js:748]: Good catch, updated where I could

**Comment by alxnddr** [comment]
@npfitz @JesseSDevaney thanks for reviews, I updated the PR and also removed `mantine` direct imports outside of `metabase/ui`.

@luizarakaki I update the copy, thanks for clarification, please have a look. I don't show the upgrade link and the copy "Only available in certain Metabase plans." if user runs Metabase EE already, is that ok?

**Comment by luizarakaki** [comment]
Yeah, makes sense to hide for pro/ee

**Comment by vbenedetti** [comment]
@alxnddr - I just checked the branch, the upgrade we did for the permissions help looks great. I'm very happy that it's coming live. 
I know this is not explicitly said on the spec, but playing with the branch, I wish the behavior when navigating from Data to Collections and Application Permissions affected the help differently.

Current: In Data Permissions I open the Permissions help, then I Click on collections. The permission help closes and I have to open again to notice the contextual change.

preferred: When clicking on Collections, if the help is open, it remains open and the content changes to match the new context. 

Could we make this little update?
Thanks!

**Review by JesseSDevaney — APPROVED** [review]
:gem: Looking good! I just have a few questions about variable naming.

**Comment by luizarakaki** [comment]
Played with the branch today and it is looking good. I'd like to see the improvement @vbenedetti proposed. It makes sense to keep the sidebar open while we change tabs.

**Review by vbenedetti — APPROVED** [review]
Looks great! thanks for adding the last minute improvement.

**Pull request merged**